### ISSUE TITLE
Improve path support on textPath

### ISSUE THREAD

**Issue by awerlang** [alafr/SVG-to-PDFKit#30]
`<textPath>` could support a few more patterns for paths.

1. Inline `d` attribute;
2. Data URIs on `href`;
3. Paths with not-only fragments (i.e. /page.html#fragment)

Item 3 is a bit more complicated, given there's no document context. How should we resolve same-document references? External reference may be trickey as well.

I would have a try at these, but would like to know what do you think.

**Comment by alafr** [comment]
I have never used any SVG with these patterns and for me it's too difficult and not useful enough to implement them. However if you really need this functionality you can:
1) edit the [`SvgElem.resolveUrl()`](https://github.com/alafr/SVG-to-PDFKit/blob/master/source.js#L798) function
2) parse the DataURIs, load the external files and merge everything into a single SVG file before using SVG-to-PDFKit (just make sure all the `id`s remain unique). This option is probably easier to implement

**Comment by awerlang** [comment]
Hello!

I'm sorry it took so long for a follow up. We went with your option 1. I saw you published to npm (that's great!) and now it would be a good time to find a better solution, because every time we updated we had to re-apply the change.

So I'll make a PR to fix this. It should work on both server (no document context) and browser. I think exposing resolveUrl as an option should work great. This will fix the issue when there's no document context, we just let the user resolve. Other option would be pass a document context for the document. I'm not sure if data URIs will be needed, but if the specs allow inlining the path that should be a better solution as well.
